### DISCUSSION TITLE
Spades metaviral assembler (v 3.15)

### DISCUSSION THREAD

**Discussion by aretchless** [LANL-Bioinformatics/EDGE#59] [Ideas]
Hello. We are testing some metagenomic strategies for identifying viruses and noticed that the default assembler (IDBA_UD) has trouble assembling the viruses. For instance, we sometimes get the full genome when mapping contigs to a reference genome-- but in thousands of contigs with extensive overlap of the contigs. I'm planning to test if the SPADES assembly does any better, but I noticed that a newer version of SPADES ([v 3.15](https://cab.spbu.ru/files/release3.15.4/changelog.html)) has a new algorithm specifically for [assembling viral genomes from metagenomic samples](https://pubmed.ncbi.nlm.nih.gov/32413137/). Before digging in, I wanted to check if you are thinking of adding this to EDGE or have done any testing. Thanks!

**Comment by aretchless** [comment]
FWIW, the [Viralverify tool ](https://github.com/ablab/viralVerify) looks interesting too (described in the viral assembly paper linked above)

> We thus developed the VIRALVERIFY tool that examines the gene content of a contig and classifies it as viral/bacterial/uncertain using a Naive Bayesian classifier. It can be used as a standalone tool to predict contigs of viral origin in any assembled metagenome.

**Reply by chienchi** [reply]
Thanks for the great suggestion. We have put this as an enhancement feature list.

**Reply by aretchless** [reply]
update: Viralverify is actually a separate package (available via Conda): https://github.com/ablab/viralVerify

Also, I've been running the various spades algorithms on our test data (with known viruses in them), and it has not gone as smoothly as I'd hoped. SPAdes crashed a fair number of times while running any of the algorithms-- as it has always tended to do (but usually completes when rerun with --continue).

The metaviral algorithm seems to have 'all or none' behavior. It returns few contigs, and often has no contigs that match the reference genome for the virus in the specimen, even when the other SPADES algorithms make very good assemblies of the virus. The contigs it returns are not necessarily for our virus -- but when one does match our reference genome, it often coverers a large portion of the genome.

The spades metagenomic algorithm seems to be the best for our data, especially when we use an enrichment kit (e.g. Twist pan-respiratory panel). I see that's already an option in EDGE, so I'll try working with that.

**Reply by chienchi** [reply]
Hi, 
We have Spades software updated to latest version 3.15.5 (see [commit 8f1003b](https://github.com/LANL-Bioinformatics/EDGE/commit/8f1003b4cab528f9b6319825eba41ee8ff61f650) ) and expand the GUI drop menu with corresponding assembly algorithms. 
![Screenshot 2023-04-18 at 3 21 18 PM](https://user-images.githubusercontent.com/737589/232908027-a4de9af8-93f7-48e7-ab25-489a14b24b32.png)
We also regenerate a [docker image](https://hub.docker.com/r/bioedge/edge_24_1_ubuntu18) along with [many other changes](https://github.com/LANL-Bioinformatics/EDGE/blob/devel/ChangeLog.txt).  It could be a pre-release EDGE version of 2.4.1 and we are testing it (the [docker image](https://hub.docker.com/r/bioedge/edge_24_1_ubuntu18)). Feel free to try it out, we appreciate any feedback to us.

Thanks.
